rwengine/src/audio/SoundSource.cpp

```diff
@@ -121,7 +121,7 @@ bool SoundSource::findAudioStream(const std::filesystem::path& filePath) {
     }
 
     audioStream = formatContext->streams[streamIndex];
-    codec = avcodec_find_decoder(audioStream->codecpar->codec_id);
+    auto codec = avcodec_find_decoder(audioStream->codecpar->codec_id);
```

Comment:
This shadows the [SoundSource::codec](https://github.com/rwengine/openrw/blob/0f83c16f6518c427a4f156497c3edc843610c402/rwengine/src/audio/SoundSource.hpp#L93).

I think the [SoundSource::codec](https://github.com/rwengine/openrw/blob/0f83c16f6518c427a4f156497c3edc843610c402/rwengine/src/audio/SoundSource.hpp#L93) should be changed to `const AVCodec*` as I'm pretty sure shadowing it only causes other issues.

Reply:
Yes I think the `SoundSource::codec` should be made `const`.

Reply:
So `const codec` instead of `auto codec`?

Reply:
No because that still shadows the member variable.

The `SoundSource::codec`: https://github.com/rwengine/openrw/blob/0f83c16f6518c427a4f156497c3edc843610c402/rwengine/src/audio/SoundSource.hpp#L93

should be:

```cpp
const AVCodec* codec = nullptr;
```

Reply:
Do you want me to make that change and commit it into this pr?

Reply:
Yes commit it to this PR. And I think you can drop the original commit.